controllers/secret_controller.go

```diff
@@ -171,6 +171,12 @@ func (r *SecretReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 		escapedPubKeyAnnotation := strings.Replace(nodeconfig.PubKeyHashAnnotation, "/", "~1", -1)
 		patchData := fmt.Sprintf(`[{"op":"add","path":"/metadata/annotations/%s","value":""}]`, escapedPubKeyAnnotation)
 		for _, node := range nodes.Items {
+			// Only clear the annotation for Nodes associated with Machines, as the clearing of the annotation is used
+			// solely for to kick off the deletion and recreation of Machines.
```

Comment:
`used soley to kick...`

Won't we need to do this for BYOH instances that need to be reconfigured?